src/DynamoCore/Graph/Nodes/CodeBlockNode.cs

```diff
@@ -239,6 +239,73 @@ public string Code
             private set { code = value; }
         }
 
+        /// <summary>
+        ///     Contains data needed to reconstruct input and output ports during deserialization 
+        ///     when there is an error in a code block node.
+        /// </summary>
+        public class ErrorStatePortInfo
+        {
+            internal List<string> inPortNames;
+            internal List<int> outPortLineIndexes;
+
+            internal ErrorStatePortInfo(
+                IEnumerable<PortModel> inPorts,
+                IEnumerable<PortModel> outPorts)
+            {
+                inPortNames = new List<string>();
+                foreach (var inPort in inPorts)
+                    inPortNames.Add(inPort.Name);
```

Comment:
do you mind using curly braces around foreach loops? we don't have anything about it in the style guide so will leave this up to you, just thought I would mention it does not seem too common in the repo.